pkg/nsx/services/subnet/subnet.go

```diff
@@ -65,20 +65,39 @@ func InitializeSubnetService(service common.Service) (*SubnetService, error) {
 		},
 	}
 
+	// Use sync.Once to ensure channel is closed only once
```

Comment:
I would prefer to change like this,

```
func InitializeSubnetService(service common.Service) (*SubnetService, error) {
	wg := sync.WaitGroup{}
	fatalErrors := make(chan error, 1)
	defer close(fatalErrors)
	subnetService := &SubnetService{
		Service: service,
		SubnetStore: &SubnetStore{
			ResourceStore: common.ResourceStore{
				Indexer: cache.NewIndexer(keyFunc, cache.Indexers{
					common.TagScopeSubnetCRUID:    subnetIndexFunc,
					common.TagScopeSubnetSetCRUID: subnetSetIndexFunc,
					common.TagScopeVMNamespace:    subnetIndexVMNamespaceFunc,
					common.TagScopeNamespace:      subnetIndexNamespaceFunc,
				}),
				BindingType: model.VpcSubnetBindingType(),
			},
		},
	}

	wg.Add(1)
	go subnetService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeSubnet, nil, subnetService.SubnetStore)
	wg.Wait()

	if len(fatalErrors) > 0 {
		err := <-fatalErrors
		return subnetService, err
	}

	return subnetService, nil
}
```

Reply:
https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/nsx/services/securitypolicy/firewall.go#L116-L116
The initial reason is to return as early as possible if there are multiple resources.
You should consider closing channel `fatalErrors` in this way.

Reply:
Yes, this version has no zombie routine.